internal/resources/webform/webform.js

```diff
@@ -2105,7 +2105,22 @@ window.initGRPCForm = function(services, svcDescs, mtdDescs, invokeURI, metadata
     function updateCurlCommand(requestDataJson) {
         let service = $("#grpc-service").val();
         let method = $("#grpc-method").val();
-        gRPCurlTextArea.html(`<div>grpcurl -plaintext -d '${requestDataJson}' ${window.target} ${service}.${method}</div>`);
+
+        var metadataStr = "";
+        var rows = $("#grpc-request-metadata-form tr");
+        for (var i = 0; i < rows.length; i++) {
+            var cells = $("input", rows[i]);
+            if (cells.length === 0) {
+                continue;
+            }
+            var name = $(cells[0]).val();
+            var val = $(cells[1]).val();
+            if (name !== "") {
+                metadataStr += ` -H "${name}: ${val}"`;
+            }
+        }
```

Comment:
One thing I noticed, changing keys/values/rows in the headers doesn't trigger `updateCurlCommand`.  I'm not sure if it's worth the level of invasiveness that would require tho.  Thoughts?

Maybe we could always just force a re-render when the Raw Request tab is shown instead of the crazy stuff we're currently doing.

Reply:
For example, I noticed a bug where on first render, if you load the app (or change the selected method) and click on the raw request tab, the curl command contains an `[object Object]` instead of `{}`.

Reply:
Yes. I agree with you. For example, the updateCurlCommand func receives the entire request instead just the body. I believe this would involve a larger PR of refactoring.  But for now, due to lack of time, I can't do that. Merging this PR would help people here a lot.

Reply:
fair enough